queries/gdscript/highlights.scm

```diff
@@ -123,4 +155,186 @@
   "puppetsync"
 ] @keyword
 
-"return" @keyword.return
+"func" @keyword.function
+
+[
+  "await"
+  "return"
+] @keyword.return
+
+(call (identifier) @keyword.return
+      (#eq? @keyword.return "yield"))
+
+
+;; Builtins
+; generated from
+; - https://github.com/godotengine/godot/blob/491ded18983a4ae963ce9c29e8df5d5680873ccb/doc/classes/@GlobalScope.xml
+; - https://github.com/godotengine/godot/blob/491ded18983a4ae963ce9c29e8df5d5680873ccb/modules/gdscript/doc_classes/@GDScript.xml
+; some from:
+; - https://github.com/godotengine/godot-vscode-plugin/blob/0636797c22bf1e23a41fd24d55cdb9be62e0c992/syntaxes/GDScript.tmLanguage.json
+
+;; Bulitin Annotations
+
+((annotation (identifier) @attribute)
+ (#any-of? @attribute
+  ; @GDScript
+  "export" "export_category" "export_color_no_alpha" "export_dir"
+  "export_enum" "export_exp_easing" "export_file" "export_flags"
+  "export_flags_2d_navigation" "export_flags_2d_physics"
+  "export_flags_2d_render" "export_flags_3d_navigation"
+  "export_flags_3d_physics" "export_flags_3d_render" "export_global_dir"
+  "export_global_file" "export_group" "export_multiline" "export_node_path"
+  "export_placeholder" "export_range" "export_subgroup" "icon" "onready"
+  "rpc" "tool" "warning_ignore"
+ ))
```

Comment:
Are all annotation identifiers not `@attribute`? Is the annotation itself not entirely `@attribute` as well? I'll play around with these changes shortly..

Reply:
I was thinking about that. There's only a handful of annotations, you can't define your own, and they are hard coded in Godot's Gdscript parser. Maybe the highlighting can show when you have a typo at glance?

> Is the annotation itself not entirely @attribute as well?

I followed queries from java: https://github.com/nvim-treesitter/nvim-treesitter/blob/3fe80dbecdf1fa88cf81a4d7b30ab0714f8c443a/queries/java/highlights.scm#L37-L40

and python: https://github.com/nvim-treesitter/nvim-treesitter/blob/3fe80dbecdf1fa88cf81a4d7b30ab0714f8c443a/queries/python/highlights.scm#L83-L87

... which I see now that python does "@" with a priority.

I'll mark the whole annotation as `@attribute`.

Reply:
While it's good to gain understanding and learn the standard from other queries, it's best not to think of it as a "one size fits all" when there are rules with similar names/functionality :)

Reply:
That's good to know that the queries should be tailored specifically for the language. I added `@attribute` to the `"@"` and kept the any-of check.